lib/ansible/modules/cloud/openstack/os_image.py

```diff
@@ -132,8 +164,13 @@ def main():
         ramdisk=dict(default=None),
         kernel=dict(default=None),
         properties=dict(type='dict', default={}),
+        volume=dict(default=None),
         state=dict(default='present', choices=['absent', 'present']),
     )
+
+    required_one_of = [['filename', 'volume']],
```

Comment:
Filename wasn't required before. Why should be required now?

Reply:
You're right, should be just mutually exclusive.